src/hooks/http.ts

```diff
@@ -13,13 +15,77 @@ export default function httpHook(mod: HTTP) {
       return target.apply(thisArg, argArray).prependListener("request", handleRequest);
     },
   });
+
+  hookClientRequestApi(mod);
+
   return mod;
 }
 
 httpHook.applicable = function (id: string) {
   return ["http", "https", "node:http", "node:https"].includes(id);
 };
 
+function hookClientRequestApi(mod: HTTP) {
+  const createApplyProxy = (f: HTTP["get" | "request"]) => {
+    return new Proxy(f, {
+      apply(target, thisArg, argArray: Parameters<typeof f>) {
+        const clientRequest = target.apply(thisArg, argArray);
+        handleClientRequest(clientRequest);
+        return clientRequest;
+      },
+    });
+  };
+
+  mod.request = createApplyProxy(mod.request);
+  mod.get = createApplyProxy(mod.get);
+
+  if ("ClientRequest" in mod) {
+    mod.ClientRequest = new Proxy(mod.ClientRequest, {
+      construct(target, argArray, newTarget): object {
+        const clientRequest = Reflect.construct(target, argArray, newTarget) as ClientRequest;
+        handleClientRequest(clientRequest);
+        return clientRequest;
```

Comment:
Doesn't this make the proxy for `get` and `request` unnecessary? I'm pretty sure those methods construct a ClientRequest.

Reply:
Looks like they are shared/reused in node binary (C++) level, not in JavaScript level, so we need to proxy them too.